catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/FileSystemPersistenceProvider.java

```diff
@@ -53,50 +56,34 @@
 
   private static final String SER_REGEX = "\\.ser";
 
-  private String mapName = "default";
+  @VisibleForTesting static final String STORAGE_ROOT_PROPERTY = "storage";
 
-  FileSystemPersistenceProvider(String mapName) {
+  private Properties properties;
+
+  private Path defaultPersistencePath;
+
+  private Path persistencePath;
+
+  FileSystemPersistenceProvider(String mapName, Properties properties) {
     LOGGER.trace("INSIDE: FileSystemPersistenceProvider constructor,  mapName = {}", mapName);
-    this.mapName = mapName;
-    File dir = new File(getPersistencePath());
-    if (!dir.exists()) {
-      boolean success = dir.mkdir();
+    this.properties = properties;
+    defaultPersistencePath = Paths.get("data", mapName);
+    persistencePath = generatePersistencePath();
+    if (!persistencePath.toFile().exists()) {
+      boolean success = persistencePath.toFile().mkdir();
       if (!success) {
-        LOGGER.info("Could not make directory: {}", dir.getAbsolutePath());
+        LOGGER.info("Could not make directory: {}", persistencePath.toFile().getAbsolutePath());
       }
     }
   }
 
-  /**
-   * Retrieve root directory of all persisted Hazelcast objects for this cache. The path is relative
-   * to containing bundle, i.e., DDF install directory.
-   *
-   * @return the path to root directory where serialized objects will be persisted
-   */
-  String getPersistencePath() {
-    return new AbsolutePathResolver("data").getPath();
-  }
-
-  /**
-   * Path to where persisted Hazelcast objects will be stored to disk.
-   *
-   * @return
-   */
-  String getMapStorePath() {
-    return Paths.get(getPersistencePath(), mapName).toString() + File.separator;
-  }
-
   @Override
   public void store(String key, Object value) {
-    File dir = new File(getMapStorePath());
-    if (!dir.exists() && !dir.mkdir()) {
-      LOGGER.debug("Unable to create directory: {}", dir.getAbsolutePath());
-    }
     LOGGER.trace("Entering: store - key: {}", key);
-    try (OutputStream file = new FileOutputStream(getMapStoreFile(key));
+    try (OutputStream file = new FileOutputStream(getMapStoreFilePath(key).toFile());
```

Comment:
❓Any idea why we're not just chaining up the constructors instead of creating temp variables we don't use?

Reply:
Only because I hadn't changed it yet